### PR TITLE
Fix the bug header bean validation is not performed

### PR DESCRIPTION
**Pull request by fujigon**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@bbdouglas @JFCote @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger 
@wing328 

### Description of the PR
this patch fixes https://github.com/OpenAPITools/openapi-generator/issues/1659

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaJaxRS/beanValidationHeaderParams.mustache [hunk 1]
```diff
@@ -0,0 +1 @@
+{{#required}} @NotNull{{/required}}{{>beanValidationCore}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaJaxRS/beanValidationHeaderParams.mustache]: @fujigon just a heads up that I may update this later to support `nullable` as well

Comment (wing328) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/beanValidationHeaderParams.mustache]: UPDATE: I've filed https://github.com/OpenAPITools/openapi-generator/pull/1679

Comment (fujigon) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/beanValidationHeaderParams.mustache]: thanks for following up!

Comment (wing328) [reply modules/openapi-generator/src/main/resources/JavaJaxRS/beanValidationHeaderParams.mustache]: @fujigon #1679 has been merged into master.

**Comment by wing328** [comment]
@fujigon thanks for the contribution, which has been included in the 4.0.0-beta release: https://twitter.com/oas_generator/status/1079727020374806529.

Happy New Year and looking forward to more collaboration and contributions in 2019!

**Pull request merged**